source/helpers/Misc.ts

```diff
@@ -7,3 +9,56 @@ export function wait(ms: number): Promise<void> {
 export function getScreenHeightProportion(proportion: number): number {
   return Math.floor((Dimensions.get("window").height * proportion) / 100);
 }
+
+export function getAPIEnvironmentIdentifierFromUrl(
+  apiUrl: string
+): string | null {
```

Comment:
How about returning an empty string instead of null?
In that way we do not need to have a return type of string | null, and we don't need to do nullish coershen (`??`) below.

Reply:
I think it signals intent better as `null`, meaning something like "there is no environment identifier for this URL". An empty string is ambiguous (is the identifier an empty string, or did we not found one?) which MAY be relevant.

`??` is then used (as opposed to `||`) since it only checks `null` and `undefined` (not "falsy"), so that the fallback is only used if we specifically didn't find an identifier, just that in this case we fallback to an empty identifier ("") as well.